errors.go

```diff
@@ -261,9 +261,12 @@ func Cause(err error) error {
 	for err != nil {
 		cause, ok := err.(causer)
 		if !ok {
-			break
+			return err
+		}
+		errCause := cause.Cause()
+		if errCause == nil || errCause == err {
+			return err
 		}
-		err = cause.Cause()
```

Comment:
If you never update err, then this for loop can never finish, and it is now possible that the loop becomes a forever loop.

Reply:
heh, was just about to test.  thank you.

Reply:
Done.